apps/music_pi/lib/music_pi/player.ex

```diff
@@ -0,0 +1,125 @@
+defmodule MusicPi.Player do
+  @moduledoc """
+  Responsible for playing music using omxplayer.
+  You can specify music directory.
+  """
+  use GenServer
+  require Logger
+
+  defmodule Data do
+    @moduledoc false
+    defstruct [:songs_path, :player_exec, :player_opts, :player_pid]
+    
+    def defaults do
+      %__MODULE__{
+        songs_path: :code.priv_dir(:music_pi),
+        player_exec: "omxplayer",
+        player_opts: "-o local",
+        player_pid: nil
+      }
+    end
+
+    def merge(opts) do
+      case Keyword.get(opts, :songs_path) do
+        nil -> defaults()
+        val -> %Data{defaults() | songs_path: val}
+      end
+    end
+  end
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, [name: __MODULE__])
+  end
+
+  def actions do
```

Comment:
It would be nice to spec `actions/0` and `run_anction/1`. The `action_id` is expected to be a `String.t`. See https://github.com/lambdaacademy/robots/blob/music-pi/apps/bot/lib/bot/service.ex for more info.